src/Catalog/Retry.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.Indexing
+{
+    public static class Retry
+    {
+        public static void Incremental(Action runLogic, Func<Exception, bool> shouldRetry, int maxAttempts, TimeSpan waitIncrement)
+        {
+            for (int currentAttempt = 0; currentAttempt < maxAttempts; currentAttempt++)
+            {
+                try
+                {
+                    runLogic();
+                    return;
+                }
+                catch (Exception e)
+                {
+                    if (currentAttempt < maxAttempts && shouldRetry(e))
```

Comment:
These could be `when` statements on the `catch`

Reply:
Sure. I just mimicked the existing synchronous implementation but I'm okay with making this change to both.

Reply:
Ah I didn't realize this file was moved. It's fine either ways 😄

Reply:
Done.